wdn/templates_6.0/js-src/components/unl-alert.js

```diff
@@ -42,6 +39,61 @@ export default class UNLAlert {
         );
     }
 
+    // Retrieves the value of a specified cookie by its name and JSON parses it.
+    #getCookieValue(name) {
+        const cookies = document.cookie.split('; ');
+        for (const cookie of cookies) {
+            const [key, value] = cookie.split('=');
+            if (key === name) {
+                return JSON.parse(decodeURIComponent(value));
+            }
+        }
+        return null;
+    }
+
+    // Sets a cookie to confirm that the 'unlalerts' JS was successfully loaded and data is received. 
+    #setAlertDataReceivedCookie(unlAlertsDataAlert) {
+        setCookie(
+            this.alertDataReceivedCookieName,
+            encodeURIComponent(JSON.stringify(unlAlertsDataAlert)),
+            this.alertDataReceivedCookieMaxAge,
+        );
+    }
+
+    async #checkIfFetchCallNeeded() {
```

Comment:
If I have two tabs open and if there is no data in unlcap.js then both tabs will be fetching the data. If there is data in unlcap.js then it preforms as intended. This is leading to unneeded requests for unlcap.js. If there is no data in unlcap.js then the cookie needs to still be set so that these unneeded requests are not being sent